plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/standard/StandardDeployJob.java

```diff
@@ -168,54 +169,175 @@ protected void canceling() {
                           .appCommandMetricsEnvironment(CloudToolsInfo.METRICS_NAME)
                           .appCommandMetricsEnvironmentVersion(CloudToolsInfo.getToolsVersion())
+                          .appCommandOutputFormat("json")
                           .build();
     return cloudSdk;
   }
 
+  private IStatus openAppInBrowser() {
+    final String project = deployConfiguration.getProject();
+    String appLocation = null;
+    if (deployConfiguration.getPromote()) {
+      appLocation = "http://" + project + ".appspot.com";
+    } else {
+      try {
+        String version =  getDeployedAppVersion();
+        appLocation = "http://" + version + "-dot-" + project+ ".appspot.com/";
+      } catch (IndexOutOfBoundsException | JsonParseException ex)  {
+        return StatusUtil.error(getClass(), Messages.getString("brower.launch.failed"), ex);
+      }
+    }
+
+    final String finalAppLocation = appLocation;
+    final IWorkbench workbench = PlatformUI.getWorkbench();
+    Job launchBrowserJob = new UIJob(workbench.getDisplay(), "Launch deployed app in browser") {
+
+      @Override
+      public IStatus runInUIThread(IProgressMonitor monitor) {
+        try {
+          URL url = new URL(finalAppLocation);
+          IWorkbenchBrowserSupport browserSupport = workbench.getBrowserSupport();
+          int style = IWorkbenchBrowserSupport.LOCATION_BAR
+              | IWorkbenchBrowserSupport.NAVIGATION_BAR | IWorkbenchBrowserSupport.STATUS;
+          browserSupport.createBrowser(style, project, project, project).openURL(url);
+        } catch (PartInitException ex) {
+          // Unable to use the normal browser support, so push to the OS
+          logger.log(Level.WARNING, "Cannot launch a browser", ex);
+          Program.launch(finalAppLocation);
+        } catch (MalformedURLException ex) {
+          return StatusUtil.error(getClass(), Messages.getString("invalid.url"), ex);
+        }
+        return Status.OK_STATUS;
+      }
+
+    };
+    launchBrowserJob.schedule();
+    return Status.OK_STATUS;
+  }
+
+  /**
+   * @return the error message obtained from <code>config.getErrorMessageProvider()</code> or
+   * <code>defaultMessage</code>
+   */
+  private String getErrorMessageOrDefault(String defaultMessage) {
+    // TODO: Check the assumption that if there are error messages during staging collected via
+    // the errorCollectingLineListener, the staging process will have a non-zero exitcode,
+    // making it ok to use the same errorCollectingLineListener for the deploy process
+    List<String> messages = errorCollectingLineListener.getCollectedMessages();
+    if (!messages.isEmpty()) {
+      return Joiner.on('\n').join(messages);
+    } else {
+      return defaultMessage;
+    }
+  }
+
+  private String getDeployedAppVersion() {
+    String rawDeployOutput = stdoutLineListener.getOutput().substring(stagingOutputEndIndex);
+    DeployOutput deployOutput = AppEngineDeployUtil.parseDeployOutput(rawDeployOutput);
```

Comment:
With above suggested changes this becomes simply 

DeployOutput.parse(rawDeployOutput)